src/courseware/course/sidebar/sidebars/course-outline/UnitIcon.jsx

```diff
@@ -40,7 +40,11 @@ const UnitIcon = ({ type, isCompleted, ...props }) => {
     },
   };
 
-  const Icon = iconMap[type]?.[isCompleted ? 'complete' : 'default'];
+  let Icon = iconMap[type || UNIT_ICON_TYPES.other];
+
+  if (typeof iconMap[type || UNIT_ICON_TYPES.other] === 'object') {
+    Icon = iconMap[type || UNIT_ICON_TYPES.other]?.[isCompleted ? 'complete' : 'default'];
+  }
```

Comment:
```suggestion
const defaultIconType = type || UNIT_ICON_TYPES.other;
let Icon = iconMap[defaultIconType];

if (typeof Icon === 'object') {
  const iconName = isCompleted ? 'complete' : 'default';
  Icon = Icon[iconName];
}
```

To improve the readability slightly and avoid a code duplication

Reply:
Agree with you. Simplified.